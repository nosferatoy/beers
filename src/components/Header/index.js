import React from 'react';
import './index.scss';

export const DEFAULT_HEADER_BG_COLOR = '#000000';

const Header = ({
	leftIcon,
	rightIcon,
	backgroundColor = DEFAULT_HEADER_BG_COLOR,
	children
}) => {
	return (
		<header className="header" style={{ backgroundColor }}>
			<div className="header__icon header__icon--left">{leftIcon}</div>
			<div className="header__container">{children}</div>
			<div className="header__icon header__icon--right">{rightIcon}</div>
		</header>
	);
};

export default Header;
