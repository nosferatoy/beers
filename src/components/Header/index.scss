$transition-time: 0.3s;

.header {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  height: var(--header-height);
  will-change: transform;
  z-index: 10;
  width: 100%;
  padding: 10px 0;
  display: flex;
  align-items: center;
  contain: content;

  &__container {
    flex: auto;
    transition: $transition-time all;
    text-align: center;
    font-weight: bold;
  }

  &__icon {
    transition: $transition-time linear;
    will-change: transform;
    color:var(--color-primary);

    &--left {
      text-align: left;
    }

    &--right {
      text-align: right;
    }
  }
}

