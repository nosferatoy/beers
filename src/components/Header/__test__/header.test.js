import React from 'react';
import { render } from '@testing-library/react';
import Header from '../index';

import logo from '../../../assets/logo.svg';

describe('Test Header component', () => {
	it('Render color', () => {
		const { container } = render(
			<Header color="#FF0000">
				<div>Test</div>
			</Header>
		);
		expect(container).toMatchSnapshot();
	});

	it('Render left icon', () => {
		const { container } = render(
			<Header
				leftIcon={
					<img
						style={{ height: '50px', width: '50px' }}
						src={logo}
						alt="logo"
					/>
				}
			>
				<div>Test</div>
			</Header>
		);
		expect(container).toMatchSnapshot();
	});

	it('Render right icon', () => {
		const { container } = render(
			<Header
				rightIcon={
					<img
						style={{ height: '50px', width: '50px' }}
						src={logo}
						alt="logo"
					/>
				}
			>
				<div>Test</div>
			</Header>
		);
		expect(container).toMatchSnapshot();
	});
});
