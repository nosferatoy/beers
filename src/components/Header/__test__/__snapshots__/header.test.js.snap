// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header component Render color 1`] = `
<div>
  <header
    class="header"
    style="background-color: rgb(0, 0, 0);"
  >
    <div
      class="header__icon header__icon--left"
    />
    <div
      class="header__container"
    >
      <div>
        Test
      </div>
    </div>
    <div
      class="header__icon header__icon--right"
    />
  </header>
</div>
`;

exports[`Test Header component Render left icon 1`] = `
<div>
  <header
    class="header"
    style="background-color: rgb(0, 0, 0);"
  >
    <div
      class="header__icon header__icon--left"
    >
      <img
        alt="logo"
        src=""
        style="height: 50px; width: 50px;"
      />
    </div>
    <div
      class="header__container"
    >
      <div>
        Test
      </div>
    </div>
    <div
      class="header__icon header__icon--right"
    />
  </header>
</div>
`;

exports[`Test Header component Render right icon 1`] = `
<div>
  <header
    class="header"
    style="background-color: rgb(0, 0, 0);"
  >
    <div
      class="header__icon header__icon--left"
    />
    <div
      class="header__container"
    >
      <div>
        Test
      </div>
    </div>
    <div
      class="header__icon header__icon--right"
    >
      <img
        alt="logo"
        src=""
        style="height: 50px; width: 50px;"
      />
    </div>
  </header>
</div>
`;
